mmv1/templates/terraform/custom_create/repository_group.go.tmpl

```diff
@@ -0,0 +1,97 @@
+userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+if err != nil {
+	return err
+}
+
+obj := make(map[string]interface{})
+repositoriesProp, err := expandGeminiRepositoryGroupRepositories(d.Get("repositories"), d, config)
+if err != nil {
+	return err
+} else if v, ok := d.GetOkExists("repositories"); !tpgresource.IsEmptyValue(reflect.ValueOf(repositoriesProp)) && (ok || !reflect.DeepEqual(v, repositoriesProp)) {
+	obj["repositories"] = repositoriesProp
+}
+labelsProp, err := expandGeminiRepositoryGroupEffectiveLabels(d.Get("effective_labels"), d, config)
+if err != nil {
+	return err
+} else if v, ok := d.GetOkExists("effective_labels"); !tpgresource.IsEmptyValue(reflect.ValueOf(labelsProp)) && (ok || !reflect.DeepEqual(v, labelsProp)) {
+	obj["labels"] = labelsProp
+}
+
+url, err := tpgresource.ReplaceVars(d, config, "{{"{{"}}GeminiBasePath{{"}}"}}projects/{{"{{"}}project{{"}}"}}/locations/{{"{{"}}location{{"}}"}}/codeRepositoryIndexes/{{"{{"}}coderepositoryindex{{"}}"}}/repositoryGroups?repositoryGroupId={{"{{"}}repository_group_id{{"}}"}}")
+if err != nil {
+	return err
+}
+
+log.Printf("[DEBUG] Creating new RepositoryGroup: %#v", obj)
+billingProject := ""
+
+project, err := tpgresource.GetProject(d, config)
+if err != nil {
+	return fmt.Errorf("Error fetching project for RepositoryGroup: %s", err)
+}
+billingProject = project
+
+// err == nil indicates that the billing_project value was found
+if bp, err := tpgresource.GetBillingProject(d, config); err == nil {
+	billingProject = bp
+}
+
+var res map[string]interface{}
+headers := make(http.Header)
+for {
+	res, err = transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+		Config:    config,
+		Method:    "POST",
+		Project:   billingProject,
+		RawURL:    url,
+		UserAgent: userAgent,
+		Body:      obj,
+		Timeout:   d.Timeout(schema.TimeoutCreate),
+		Headers:   headers,
+	})
+	if err != nil {
+		if transport_tpg.IsGoogleApiErrorWithCode(err, 409) {
```

Comment:
This assumes all 409 code errors are retryable, which is probably not the case. Could we also look into the error message or any other information to narrow down the error to the "queue is full"?

Reply:
409 is also returned in the case where the CRI is not ready yet, so we will have to include this too